ui/fancy/src/js/autorepeat.js

```diff
@@ -0,0 +1,53 @@
+class KeyAutorepeater {
+    constructor(repeatDelay, repeatRate) {
+      this.repeatDelay = repeatDelay;
+      this.repeatRate = repeatRate;
+      this.listeners = new Map();
+      this.repeaters = new Set();
+      this.keyDownData = new Map();
+      window.addEventListener('keydown', (ev) => {
```

Comment:
You really want to install these listeners in constructor? There should be separate method for it.